### PR TITLE
Deal with app config ports

### PR DESCRIPTION
**Pull request by suprjinx**

### CODE REVIEW

File: apis/install/v1alpha1/common.go [hunk 1]
```diff
@@ -93,4 +100,37 @@ type CommonSpecBase struct {
...
+}
+
+// BuildPortConfig extracts ports from the ApplicationConfig and applies inplace
+// as the PortConfig
+func (c *CommonSpecBase) BuildPortConfig() error {
+	appConfig, err := convertRawExtensionToYaml(c.ApplicationConfig)
+	if err != nil {
+		return err
+	}
+	// defaults
```
Comment (kannon92) [inline apis/install/v1alpha1/common.go:114]: We probably shouldn't have defaults here.

Comment (suprjinx) [reply apis/install/v1alpha1/common.go:114]: why not?

File: apis/install/v1alpha1/common_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestBuildPortConfig(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    runtime.RawExtension
+		expected PortConfig
+		wantErr  bool
```
[the remark this answers is not in the document]
Comment (suprjinx) [reply apis/install/v1alpha1/common_test.go:16]: added one, but invalid app config will cause errors everywhere

File: apis/install/v1alpha1/common.go (hunk 1 above)
Comment (suprjinx) [reply apis/install/v1alpha1/common.go:114]: Also, the controller for a given service creates the port configuration -- this is just a container for storing it. so having an httpPort default will not expose that automatically unless the controller is explicitly creating an http Deployment, Service, etc.

File: apis/install/v1alpha1/common_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kannon92) [reply apis/install/v1alpha1/common_test.go:16]: yea its mostly for coverage for this function and not for the entire app.

**Review by kannon92 — APPROVED** [review]

**Pull request merged**